src/block-components/progress-bar/style.js

```diff
@@ -75,12 +87,42 @@ const getStyleParams = () => {
 				const circumference = Math.PI * ( radius * 2 )
 				return ( ( 100 - derivedPercent ) / 100 ) * circumference
 			},
+		} ] : [ {
+			selector: '[class*="stk-progress-"]',
+			styleRule: '--progress-border-radius',
+			attrName: 'progressBorderRadius',
+			hasUnits: 'px',
+		},
+		{
+			selector: '.stk-progress__bar.stk--has-background-overlay:before',
+			styleRule: 'backgroundImage',
+			attrName: 'progressColor1',
+			enabledCallback: getAttribute => getAttribute( 'progressColorType' ) === 'gradient',
+			valueCallback: ( value, getAttribute ) => {
+				if ( ! getAttribute( 'progressColor2' ) ) {
+					return undefined
+				}
+				// The default color is the same as the other one but transparent. Same so that there won't be a weird transition to transparent.
+				const defaultColor1 = hexToRgba( getAttribute( 'progressColor2' ) || '#ffffff', 0 )
+				const defaultColor2 = hexToRgba( getAttribute( 'progressColor1' ) || '#3498db', 0 )
+
+				// Gradient location.
+				const color1Location = `${ getAttribute( 'progressColorGradientLocation1' ) || '0' }%`
+				const color2Location = `${ getAttribute( 'progressColorGradientLocation2' ) || '100' }%`
+
+				const directionAngle = getAttribute( 'progressColorGradientDirection' )
+				const angle = typeof directionAngle === 'string' ? '90deg' : `${ directionAngle }deg`
+
+				return `linear-gradient(${ angle }, ${ getAttribute( 'progressColor1' ) || defaultColor1 } ${ color1Location }, ${ getAttribute( 'progressColor2' ) || defaultColor2 } ${ color2Location })`
+			},
+			dependencies: [ 'progressColorType', 'progressColor1', 'progressColor2', 'progressColorGradientLocation1', 'progressColorGradientLocation2', 'progressColorGradientDirection' ],
 		},
+	 ] ),
 	]
 }
 
 export const Style = props => {
-	const progressBarStyles = useStyles( getStyleParams() )
+	const progressBarStyles = useStyles( getStyleParams( { isCircle: props.isCircle } ) )
```

Comment:
I think we can pass the props directly here.

```suggestion
	const progressBarStyles = useStyles( getStyleParams( props ) )
```